tools/wptserve/wptserve/request.py

```diff
@@ -262,22 +266,16 @@ def __init__(self, request_handler):
         self.url_base = "/"
 
         if self.request_path.startswith(scheme + "://"):
-            self.url = request_handler.path
+            self.url = self.request_path
         else:
-            self.url = "%s://%s:%s%s" % (scheme,
-                                      host,
-                                      port,
-                                      self.request_path)
+            self.url = "%s://%s:%s%s" % (
+                scheme, host, port, self.request_path)
         self.url_parts = urlsplit(self.url)
 
-        self.raw_headers = request_handler.headers
-
         self.request_line = request_handler.raw_requestline
 
-        self._headers = None
-
         self.raw_input = InputFile(request_handler.rfile,
-                                   int(self.headers.get("Content-Length", 0)))
+                                   int(self.raw_headers.get("Content-Length", 0)))
```

Comment:
How sure are we that the fact that `raw_headers` is case-sensitive doesn't matter here? Per the RFC header names are case-insensitive: https://tools.ietf.org/html/rfc7230#section-3.2

Reply:
I think the raw headers from the base handler are already case-insenstive.

Python 3: HTTPMessage inherits from https://docs.python.org/3/library/email.compat32-message.html#email.message.Message and
> Headers are stored and returned in case-preserving form, but field names are matched case-insensitively.

Python 2: I can't find definitive docs about this, but the implementation of the base `rfc822.Message` class does case-insensitve matching, too: https://github.com/python/cpython/blob/8d21aa21f2cbc6d50aab3f420bb23be1d081dac4/Lib/rfc822.py#L393
Since Python 2 already reaches EOL, I think it's safe to assume that behaviours will never change.